### PR TITLE
MARBLE-931, add i18n to app pages and components

### PR DESCRIPTION
**Pull request by jmille79**

### CODE REVIEW

File: @ndlib/gatsby-theme-marble/src/components/App/Pages/User/UserLayout/FollowButton/index.js [hunk 1]
```diff
@@ -16,7 +17,9 @@ export const FollowButton = ({ userName, showButton, following = false }) => {
           padding: '.5rem',
         }}
       >
-        <p><Link to={`/user`}>Log in</Link> to follow users or access and edit your own content.</p>
+        <Trans i18nKey='text:userPage:followPrompt'>
+          <Link to={'/user'}>Log in</Link>.
+        </Trans>
       </div>
     )
   } else if (following) {
```
Comment (danthewolfe) [inline @ndlib/gatsby-theme-marble/src/components/App/Pages/User/UserLayout/FollowButton/index.js:25]: Don't forget the "follow" and "unfollow" in the `else` statements.

File: @ndlib/gatsby-theme-marble/src/components/App/Pages/User/UserLayout/FollowButton/index.js [hunk 2]
```diff
@@ -16,22 +18,28 @@ export const FollowButton = ({ userName, showButton, following = false }) => {
           padding: '.5rem',
         }}
       >
-        <p><Link to={`/user`}>Log in</Link> to follow users or access and edit your own content.</p>
+        <Trans i18nKey='text:userPage.followrompt'>
```
Comment (mlkirkpatrick) [inline @ndlib/gatsby-theme-marble/src/components/App/Pages/User/UserLayout/FollowButton/index.js:21]: Typo.

File: @ndlib/gatsby-theme-marble/src/assets/i18n/json/text.default.en.json [hunk 3]
```diff
@@ -6,6 +6,29 @@
   "noUserPage": "The user <1>{{ userName }}</1> does not exist or you are not authorized to view this page.",
   "userLogoutPage": "You have been successfully logged out.",
   "userPage": {
-    "loginPrompt": "<0>Log in</0> to access and edit your own content."
+    "loginPrompt": "<0>Log in</0> to access and edit your own content.",
```
Comment (mlkirkpatrick) [inline @ndlib/gatsby-theme-marble/src/assets/i18n/json/text.default.en.json:9]: I don't see this one implemented. Just making sure it wasn't missed.

File: @ndlib/gatsby-theme-marble/src/components/App/Pages/Portfolio/PortfolioBody/PortfolioEditSettings/PortfolioSettingsContent/VisibilitySettings/index.js [hunk 4]
```diff
@@ -2,22 +2,24 @@ import React from 'react'
...
+    {
+      value: t('text:portfolioPage.visibilityPrivate.value'),
+      description: t('text:portfolioPage.visibilityPrivate.description'),
+    },
+    {
+      value: t('text:portfolioPage.visibilityShared.value'),
+      description: t('text:portfolioPage.visibilityShared.description'),
+    },
+    {
+      value: t('text:portfolioPage.visibilityPublic.value'),
```
Comment (danthewolfe) [inline @ndlib/gatsby-theme-marble/src/components/App/Pages/Portfolio/PortfolioBody/PortfolioEditSettings/PortfolioSettingsContent/VisibilitySettings/index.js:19]: This one is a bit tricky. The label needs to be translated, but the actual internal value needs to remain the same for things to work correctly. In this case the original code was basically always assuming they were going to be the same.

What I would recommend is that we leave values here, `public`, `private`, `shared` and do the translation in the `VisibilityLabel` component that is used inside `FormattedLabel` to actually do the text rendering. By modifying `VisibilityLabel` this way it should take care of all visible occurrences.

File: @ndlib/gatsby-theme-marble/src/assets/i18n/json/text.default.en.json (hunk 3 above)
Comment (jmille79) [reply @ndlib/gatsby-theme-marble/src/assets/i18n/json/text.default.en.json:9]: This part was already implemented in App/Pages/User/UserLayout/PromptLogin

**Pull request merged**